### PR TITLE
AssertionEngine support mock recommended assertions

### PR DESCRIPTION
**Pull request by kentwelcome**
Signed-off-by: Kent Huang <kent@infuseai.io>

** PR checklist **

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: piperider_cli/assertion_engine/assertion.py [hunk 1]
```diff
@@ -230,47 +235,137 @@ def load_assertions(self):
...
+        elif isinstance(selected_tables, list):
+            tables = {t: tables[t] for t in selected_tables}
+
+        recommended_assertions = self._generate_assertion_template(tables)
+
+        # TODO: Generate recommended assertions
+        recommender = AssertionRecommender(recommended_assertions, profiling_result)
+        recommender.recommend()
+
+        # self._mock_recommended_table_assertions(recommended_assertions, profiling_result)
```
Comment (kentwelcome) [inline piperider_cli/assertion_engine/assertion.py:255]: Remove commented code

File: piperider_cli/assertion_engine/assertion.py [hunk 2]
```diff
@@ -230,47 +235,137 @@ def load_assertions(self):
...
+            if assertion.get('skip'):  # skip if it already exists user-defined assertions
+                continue
             with open(file_path, 'w') as f:
-                print(
-                    f'Generating assertion template for table "{table}" -> {file_path}')
                 yaml.YAML().dump(assertion, f)
+                paths.append(file_path)
+        return paths
+
+    def _mock_recommended_table_assertions(self, assertions, profiling_result):
```
Comment (kentwelcome) [inline piperider_cli/assertion_engine/assertion.py:342]: Remove unused function

File: piperider_cli/assertion_engine/assertion.py [hunk 3]
```diff
@@ -230,47 +235,137 @@ def load_assertions(self):
...
+                'name': 'assert_row_count_in_range',
+                'assert': CommentedMap({
+                    'count': [int(row_count * 0.9), int(row_count * 1.1)],
+                }),
+                'tags': ['PIPERIDER_RECOMMENDED_ASSERTION'],
+            })
+            assertion[table]['tests'].append(assert_row_count_in_range)
+        pass
+
+    def _mock_recommended_column_assertions(self, assertions, profiling_result):
```
Comment (kentwelcome) [inline piperider_cli/assertion_engine/assertion.py:356]: Remove unused function

**Review by qrtt1 — CHANGES_REQUESTED** [review]
```
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/qrtt1/temp/piperider-cli/piperider_cli/workspace.py:489 in _transform_assertion_result    │
│                                                                                                  │
│   486 def _transform_assertion_result(table: str, results):                                      │
│   487 │   tests = []                                                                             │
│   488 │   columns = {}                                                                           │
│ ❱ 489 │   for r in results:                                                                      │
│   490 │   │   if r.table == table:                                                               │
│   491 │   │   │   entry = r.to_result_entry()                                                    │
│   492 │   │   │   if r.column:                                                                   │
│                                                                                                  │
│ ╭────── locals ──────╮                                                                           │
│ │ columns = {}       │                                                                           │
│ │ results = None     │                                                                           │
│ │   table = 'ACTION' │                                                                           │
│ │   tests = []       │                                                                           │
│ ╰────────────────────╯                                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'NoneType' object is not iterable
```

got `None`

**Review by qrtt1 — APPROVED** [review]

**Pull request merged**